app/src/main/java/com/kickstarter/libs/utils/ApplicationLifecycleUtil.kt

```diff
@@ -117,17 +113,14 @@ class ApplicationLifecycleUtil(private val application: KSApplication) :
     private fun refreshUser() {
         val accessToken = currentUser.accessToken ?: ""
 
-        // Check if the access token is null and the user is still logged in.
-        if (isLoggedIn && accessToken.isNull()) {
-            forceLogout("access_token_null")
-        } else {
-            if (accessToken.isNotNull() && accessToken.isNotEmpty()) {
-                client.fetchCurrentUser()
-                    .compose(Transformers.neverErrorV2())
-                    .subscribe { user ->
-                        currentUser.refresh(user)
-                    }.addToDisposable(disposables)
-            }
+        if (accessToken.isNotNull() && accessToken.isNotEmpty()) {
+            client.fetchCurrentUser()
+                .doOnError {
+                    forceLogout(it.message ?: "")
```

Comment:
When app comes back from background refresh the current logged in user. If any error happens force a logout.